### DISCUSSION TITLE
Split table in Janusgraph - Hbase

### DISCUSSION THREAD

**Discussion by dinhphu2k1-gif** [JanusGraph/janusgraph#3398] [Q&A] [answered]
When using Hbase as storeage backend, I found that Janusgraph create 1 table in Hbase as Below:

- graphindex -> g
- graphindex_lock -> h
- janusgraph_ids -> i
- edgestore -> e
- edgestore_lock -> f
- system_properties -> s
- system_properties_lock -> t
- systemlog -> m
- txlog -> l

So when Hbase split table to regions. Assuming there are 9 regions, is each column family above 1 region?

**Comment by vtslab** [comment, marked as answer]
All user data end up in the edgestore, so the edgestore is most likely to split in regions. HBase does the split automatically, after a the first region becomes larger than 20 GB (configurable value). During the initial batch load you can set the JanusGraph property storage.hbase.region-count to increase the number of regions for performance reasons. If these regions stay below 20 GB, HBase will compact these automatically after some time.
The other tables are completely independent and HBase will treat them in an analogous way.

**Reply by dinhphu2k1-gif** [reply]
@vtslab  I'm interested are 1 region containing only 1 column family or can contain many column families

**Reply by vtslab** [reply]
A region has only data of a single column family. A region server can host regions of multiple column families.
